### PR TITLE
Add API document and its generating util

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Closes https://github.com/pingcap/tidb-operator/issues/1571

### What is changed and how does it work?
Add API document and its generating util

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
Add API document and its generating util
```

### CODE REVIEW

File: hack/update-crd-groups.sh [hunk 1]
```diff
@@ -39,3 +39,15 @@ to-crdgen generate backupschedule >> $crd_target
 to-crdgen generate tidbmonitor >> $crd_target
 to-crdgen generate tidbinitializer >> $crd_target
 to-crdgen generate tidbclusterautoscaler >> $crd_target
+
+go install github.com/ahmetb/gen-crd-api-reference-docs
```
Comment (cofyc) [inline hack/update-crd-groups.sh]: install its binary release like `hack::ensure_kubectl`, then `go.(mod/sum)` will not be touched and `go mod tidy` is not required

File: docs/api-references/docs.html [hunk 2]
```diff
@@ -0,0 +1,12489 @@
...
+<p>
+<p>Package v1alpha1 is the v1alpha1 version of the API.</p>
+</p>
+Resource Types:
+<ul><li>
+<a href="#pingcap.com/v1alpha1.Backup">Backup</a>
+</li><li>
+<a href="#pingcap.com/v1alpha1.BackupSchedule">BackupSchedule</a>
+</li><li>
+<a href="#pingcap.com/v1alpha1.DataResource">DataResource</a>
```
Comment (DanielZhangQD) [inline docs/api-references/docs.html]: Can we hide this?

Comment (Yisaer) [reply docs/api-references/docs.html]: Why?

Comment (Yisaer) [reply docs/api-references/docs.html]: updated.

File: hack/update-crd-groups.sh (hunk 1 above)
Comment (Yisaer) [reply hack/update-crd-groups.sh]: updated

File: hack/lib.sh [hunk 3]
```diff
@@ -201,3 +203,23 @@ function hack::ensure_aws_k8s_tester() {
...
+}
+
+function hack::ensure_gen_crd_api_references_docs() {
+    if hack::verify_gen_crd_api_references_docs; then
+        return 0
+    fi
+    echo "Installing gen_crd_api_references_docs $DOCS_VERSION..."
+    tmpdir=$(mktemp -d)
+    trap "test -d $tmpdir && rm -r $tmpdir" RETURN
+    curl --retry 10 -L -o ${tmpdir}/docs-bin.tar.gz https://github.com/ahmetb/gen-crd-api-reference-docs/releases/download/v0.1.5/gen-crd-api-reference-docs_darwin_amd64.tar.gz
```
Comment (cofyc) [inline hack/lib.sh]: - replace `darwin` with `${OS}`
- replace `amd64` with `${ARCH}`
- replace `0.1.5` with `${DOCS_VERSION}`

File: hack/lib.sh [hunk 4]
```diff
@@ -27,12 +27,14 @@ TERRAFORM_VERSION=${TERRAFORM_VERSION:-0.12.12}
...
 KUBECTL_BIN=$OUTPUT_BIN/kubectl
 HELM_BIN=$OUTPUT_BIN/helm
+DOCS_BIN=$OUTPUT_BIN/gen-crd-api-reference-docs
 #
 # Don't ugprade to 2.15.x/2.16.x until this issue
 # (https://github.com/helm/helm/issues/6361) has been fixed.
 #
 HELM_VERSION=${HELM_VERSION:-2.9.1}
 KIND_VERSION=${KIND_VERSION:-0.7.0}
+DOCS_VERSION=${DOCS_VERSION:-v0.1.5}
```
Comment (cofyc) [inline hack/lib.sh]:
```suggestion
DOCS_VERSION=${DOCS_VERSION:-0.1.5}
```

File: hack/update-crd-groups.sh [hunk 5]
```diff
@@ -17,6 +17,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+
```
Comment (cofyc) [inline hack/update-crd-groups.sh]:
```suggestion
```

File: hack/lib.sh [hunk 6]
```diff
@@ -201,3 +203,23 @@ function hack::ensure_aws_k8s_tester() {
 	mv $tmpfile $AWS_K8S_TESTER_BIN
 	chmod +x $AWS_K8S_TESTER_BIN
 }
+
+function hack::verify_gen_crd_api_references_docs() {
+    if test -x "$DOCS_BIN"; then
+        return
```
Comment (cofyc) [inline hack/lib.sh]:
```suggestion
        # TODO check version when the binary version is available.
        return
```

File: hack/lib.sh (hunk 3 above)
Comment (Yisaer) [reply hack/lib.sh]: updated

**Pull request merged**